File: lib/src/main/java/pl/tajchert/waitingdots/SinTypeEvaluator.java
package pl.tajchert.waitingdots;

import android.animation.TypeEvaluator;

public class SinTypeEvaluator implements TypeEvaluator<Number> {
    @Override
    public Number evaluate(float fraction, Number from, Number to) {
        return Math.max(0, Math.sin(fraction * Math.PI * 2)) * (to.floatValue() - from.floatValue());
    }
}

File: lib/src/main/java/pl/tajchert/waitingdots/SinTypeEvaluator.kt
package pl.tajchert.waitingdots

import android.animation.TypeEvaluator

class SinTypeEvaluator : TypeEvaluator<Number> {
  override fun evaluate(
    fraction: Float,
    from: Number,
    to: Number
  ): Number {
    return Math.max(0.0, Math.sin(fraction.toDouble() * Math.PI * 2.0)) * (to.toFloat() - from.toFloat())
  }
}
